engine/gamesys/src/gamesys/components/comp_collision_object.cpp

```diff
@@ -162,6 +162,11 @@ namespace dmGameSystem
 
     dmGameObject::CreateResult CompCollisionObjectNewWorld(const dmGameObject::ComponentNewWorldParams& params)
     {
+        if (params.m_MaxComponentIntances == 0)
+        {
+            *params.m_World = 0x0;
+            return dmGameObject::CREATE_RESULT_OK;
+        }
```

Comment:
it worth to avoid creation of the new world in case collection doesn't contain physics components. We will save extra memory on raycasts and the other elements.

Reply:
What happens if I call `physics.raycast()` in such a collection?

Reply:
good point, it will crash

Reply:
@britzl not sure how to solve it better. I would like to avoid physics world creation anyway but in this case I need to add protective code that checking if world exist.

Reply:
I added some checks for world == 0x0

Reply:
In general, I'd try to stop it as early as possible. E.g. in the script is possible.
For physics though, we have both messages and actual functions.
Functions are easier, since they need to get the correct world before they can call the function.

For the physics case, I think a few world==0 checks are ok.

Reply:
I stand corrected, even `Physics_RayCastAsync` uses the `world`, so you can check it.

Reply:
now i check it in `script_physics.cpp`